### ISSUE TITLE
bar(..., linewidth=None) doesn't display bar edges with mpl2.0b3

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#6894]
Regression in mpl2.0b3 compared to 1.5.1:
```
bar([0, 1], [2, 3], color="none", edgecolor="k", linewidth=None)
```
doesn't draw edges for the bars.  The docstring (which btw has a weird linebreak) states:
```
    linewidth : scalar or array-like, optional
        width of bar edge(s). If None, use default
        linewidth; If 0, don't draw edges.
        default: None
```
Perhaps due to a change in the defaults, I'm not sure?  But even if this was the case at least the docstring should be updated to indicate which rcparam gives the default, and state that this default is zero.

**Comment by tacaswell** [comment]
This is probably related to #6844

**Comment by efiring** [comment]
This is fixed by #6904.

**Comment by tacaswell** [comment]
Gives 
![so](https://cloud.githubusercontent.com/assets/199813/18065043/41071aba-6e00-11e6-8b05-255544157664.png)

on current 2.x
